src/udf/python/risingwave/udf.py

```diff
@@ -78,6 +78,17 @@ def _process_input_array(array: list, type: pa.DataType) -> list:
     return array
 
 
+def _process_output_array(array: list, type: pa.DataType) -> list:
+    if pa.types.is_list(type):
```

Comment:
Seems we need to recursive into struct (tuple) type as well?

Reply:
I didn't consider struct output before... Fixed in 8330028